### ISSUE TITLE
get iterates

### ISSUE THREAD

**Issue by hiyamgh** [ronakdm/prospect#1]
Hello, thank you very much for posting the code.

```
def get_iterates(dataset, objective, optimizer, seed=1):
    loss = "squared_error" if dataset == "acsincome" else "binary_cross_entropy"
    model_cfg = {
        "objective": objective,
        "l2_reg": l2_reg,
        "loss": loss,
        "n_class": None,
        "shift_cost": shift_cost
    }

    path = get_path([dataset, var_to_str(model_cfg), optimizer], out_path=result_dir)
    f = os.path.join(path, f"iterates_{seed}.p")
    return [iterate.detach() for iterate in pickle.load(open(f, "rb"))]
```

I am not able to generate the file `iterates_{seed}.p`, and I don't see it generated anywhere in the code. Could you please advise?

**Comment by ronakdm** [comment]
Thank you for identifying this issue. The `src/utils.training.py` file has been modified to support saving iterations (this feature was lost when preparing the final version of the package). The relevant function is copied below:
```
def train_model(optimizer, val_objective, n_epochs, save_iters=False):
    epoch_len = optimizer.get_epoch_len()
    metrics = [compute_metrics(-1, optimizer, val_objective, 0.0)]
    init_loss = metrics[0]["train_loss"]

    iterates = []
    for epoch in tqdm(range(n_epochs)):
        tic = time.time()
        optimizer.start_epoch()
        for _ in range(epoch_len):
            optimizer.step()
        optimizer.end_epoch()
        toc = time.time()
        if save_iters:
            iterates.append(optimizer.weights.clone())

        # Logging.
        metrics.append(compute_metrics(epoch, optimizer, val_objective, toc - tic))
        if metrics[-1]["train_loss"] >= 1.5 * init_loss:
            raise OptimizationError(
                f"train loss 50% greater than inital loss! (epoch {epoch})"
            )

    result = {
        "weights": optimizer.weights,
        "metrics": pd.DataFrame(metrics),
    }
    if save_iters:
        return result, iterates
    return result
```
Supply the `save_iters=True` flag above. Finally, please consider taking a look at our standalone package [DRLearn](https://ronakdm.github.io/drlearn/), which includes Prospect/LSVRG in a simplified interface.
